src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/SplitTrainingPlan.java

```diff
@@ -88,65 +93,138 @@ private Training getTrainingForDay(Map<String, List<ExerciseExecution>> training
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "BACK"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "THIGHS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "SIXPACK", "SHOULDERS"));
+                trainingList.add(getMapOfBodyPartsExercisesForDay(exercisesForBodyPart, "CALVES", "TRICEPS"));
             }
         }
+
+        return trainingList;
+    }
+
+    private Map<String, List<ExerciseExecution>> getMapOfBodyPartsExercisesForDay(Map<String, List<ExerciseExecution>> trainingForBodyPart, String... bodyParts) {
+        Map<String, List<ExerciseExecution>> map = new HashMap<>();
+        for (String part : bodyParts)
+            if (trainingForBodyPart.containsKey(part))
+                map.put(part, trainingForBodyPart.get(part));
+
+        return map;
+    }
+
+    public List<Training> normalize(List<Map<String, List<ExerciseExecution>>> list) {
+        var random = new Random();
+        int maxIndex, minIndex;
+        int min;
+        int max;
+
+        do {
+            List<Integer> listOfMapsSize = list.stream()
+                    .map(stringListMap -> stringListMap.keySet().size())
+                    .collect(Collectors.toList());
+            maxIndex = 0;
+            minIndex = 0;
+            min = list.get(0).size();
+            max = list.get(0).size();
+            for (int i = 1; i < listOfMapsSize.size(); i++) {
+                if (listOfMapsSize.get(i) > max) {
+                    maxIndex = i;
+                    max = listOfMapsSize.get(i);
+                }
+                if (listOfMapsSize.get(i) < min) {
+                    minIndex = i;
+                    min = listOfMapsSize.get(i);
+                }
+            }
+
+            if (max - min > 1) {
+                String randomBodyPart = (String) list.get(maxIndex)
+                        .keySet()
+                        .toArray()[random.nextInt(listOfMapsSize.get(maxIndex))];
+                var exerciseExecutions = list.get(maxIndex).get(randomBodyPart);
+                list.get(maxIndex).remove(randomBodyPart);
+                list.get(minIndex).put(randomBodyPart, exerciseExecutions);
+            }
+        } while (max - min > 1);
+
+
+        return parseMapOfExercisesToListOfExercises(list);
+    }
+
+    private List<Training> parseMapOfExercisesToListOfExercises(List<Map<String, List<ExerciseExecution>>> list) {
+        var trainingList = new ArrayList<Training>();
+        list.forEach(trainingDay -> {
+            Training training = new Training();
+            List<ExerciseExecution> exerciseExecutionList = new ArrayList<>();
+            trainingDay.keySet()
+                    .forEach(key -> exerciseExecutionList.addAll(trainingDay.get(key)));
+            training.setExercisesExecutions(exerciseExecutionList);
+            trainingList.add(training);
```

Comment:
Trochę długa metoda jak na lambdę, może wydzielić to do prywatnej i przekazać referencję?